python/ray/serve/_private/router.py

```diff
@@ -321,7 +321,9 @@ async def schedule_and_send_request(
             # request will be placed on the front of the queue to avoid tail latencies.
             # TODO(edoakes): this retry procedure is not perfect because it'll reset the
             # process of choosing candidates replicas (i.e., for locality-awareness).
-            replica = await self.choose_replica_for_request(pr, is_retry=True)
+            replica = await self._replica_scheduler.choose_replica_for_request(
```

Comment:
This was another small bug caused by previous refactoring. This will be covered once the FF is turned on. Also planning to add better unit testing to `router.py`